src/jit/codegenarmarch.cpp

```diff
@@ -1359,6 +1359,11 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     genConsumeAddress(tree->Addr());
     emit->emitInsLoadStoreOp(ins_Load(targetType), emitTypeSize(tree), targetReg, tree);
     genProduceReg(tree);
+
+    if (tree->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
I'm not an expert, but I believe `dmb st` is not useful for either the acquire or the release semantics of volatile.  TBH, I'm not sure what it *is* useful for, but I don't think it's this.  I believe the ARMv8 acq/rel variants of load/store instructions are exactly what we want for ARM64.  (Incidentally, we probably want them for ARM32 as well if detect their availability at runtime.)

Until we get the `ld.acq`/`st.rel` thing figured out, I think a single `dmb sy` after a load and before a store should be more than sufficient.